src/components/EffectiveDateTime/EffectiveDateTime.vue

```diff
@@ -0,0 +1,477 @@
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('selectHour')
+  onSelectHourChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('selectMinute')
+  onSelectMinuteChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('selectPeriod')
+  onSelectPeriodChanged (val: string): void {
+    if (this.isFutureEffective) {
+      this.constructAndUpdate()
+    }
+  }
+
+  @Watch('effectiveDateType')
+  onEffectiveDateTypeChanged (val: EffectiveDateTypes): void {
+    this.isImmediate = (val === EffectiveDateTypes.IMMEDIATE)
+    this.isFutureEffective = (val === EffectiveDateTypes.FUTURE_EFFECTIVE)
+
+    // if we changed to IMMEDIATE then clear the model values (otherwise retain them)
+    if (this.isImmediate) {
+      this.datePicker = ''
+      this.dateText = ''
+      this.selectHour = []
+      this.selectMinute = []
+      this.selectPeriod = PeriodTypes.AM
+      this.$refs.datePickerRef.clearDate()
+    }
+
+    // update the parent
+    this.emitIsFutureEffective(this.isFutureEffective)
+    this.emitEffectiveDate(null)
+    this.emitValid()
+  }
+
+  @Emit('isFutureEffective')
+  private emitIsFutureEffective (val: boolean): void {}
+
+  @Emit('effectiveDate')
+  private emitEffectiveDate (val: Date): void {}
+
+  @Emit('valid')
+  private async emitValid (): Promise<boolean> {
+    // localized dateText check for future effective selections
+    const validDateText = this.isFutureEffective ? !!this.dateText : true
+
+    // wait for form to update itself before checking validity
+    await Vue.nextTick()
+    const isDateValid = this.$refs.datePickerRef.validateForm()
+    const isTimeValid = this.$refs.form.validate()
+    return this.isImmediate || (!!this.effectiveDateType &&
+      isDateValid && isTimeValid &&
+      !!this.selectHour.length && !!this.selectMinute.length &&
+      !this.isUnderTime &&
+      !this.isOverTime &&
+      validDateText
+    )
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+#effective-date-time-box {
+  padding: 2rem 2rem 0.5rem;
+  line-height: 1.2rem;
+}
+
+::v-deep .v-label {
+  color: $gray7;
+  font-weight: normal;
+}
+
+.v-radio {
+  padding-bottom: .5rem;
+}
+
+.date-time-selectors {
+  margin-left: 2rem;
+}
+
+.time-colon {
+  margin-left: -4px;
+  margin-right: -4px;
+  padding-top: 2rem;
+  font-size: 25px;
+}
+
+@media (max-width: 768px) {
+  .time-colon {
+    display: none;
+  }
+}
+
+.label-col {
+  position: relative;
+  align-self: center;
+}
+
+.time-zone-label {
+  position: absolute;
+  top: -10px;
+  color: $gray7;
+}
+
+.disabled {
+  color: $gray6;
+}
+
+.validation-alert {
+  position: relative;
+
+  .validation-alert-msg {
+    line-height: 12px;
+    position: absolute;
+    top: -2rem;
+    padding: 0 12px;
+    font-size: 12px;
+    font-weight: 500;
+    color: $BCgovInputError !important;
```

Comment:
Are these defined in the shared components?

Reply:
Yes, these CSS variables are defined in https://github.com/bcgov/bcrs-shared-components/blob/main/src/assets/styles/theme.scss.

And yes, the parent app needs to have these defined as well.

I think/hope the build steps will identify any incorrect imports/variables.

Reply:
Copy that!